core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -823,11 +825,9 @@ protected final void wrapEntriesForPrepare(TxInvocationContext ctx, PrepareComma
          for (WriteCommand c : command.getModifications()) {
             c.setTopologyId(command.getTopologyId());
             // TODO: we need async invocation since the interceptors may do remote get
-            InvocationStage visitorStage = (InvocationStage) c.acceptVisitor(ctx, entryWrappingVisitor);
-            if (visitorStage != null) {
-               // Wait for the sub-command to finish. If there was an exception, rethrow it.
-               visitorStage.get();
-            }
+            InvocationStage visitorStage = makeStage(c.acceptVisitor(ctx, entryWrappingVisitor));
```

Comment:
it is not clear for me why do you need an ```InvocationStage```. We are waiting for it in the next line.